cmd/admin_build_instance.go

```diff
@@ -81,24 +78,15 @@ func getInstanceBuildRequest(l log.Logger, jobName, inputDirectory, host, projec
 	}
 	jobScheduledTimeProto := timestamppb.New(jobScheduledTime)
 
-	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
-	defer dialCancel()
-
-	var conn *grpc.ClientConn
-	if conn, err = createConnection(dialTimeoutCtx, host); err != nil {
-		if errors.Is(err, context.DeadlineExceeded) {
-			l.Error(ErrServerNotReachable(host).Error())
-		}
+	ctx, conn, closeConn, err := initClientConnection(l, host, adminBuildInstanceTimeout)
```

Comment:
Can you try pushing creation of dial context to createConnection. That context is only required at time of creating connection.

Also if err is context.DeadlineExceeded then return `ErrServerNotReachable` error.

Reply:
concerns are received from separate call, will check and update accordingly

Reply:
concerns are addressed and some bugs are fixed.